### PR TITLE
Remove remnants of Repeat executor node.

### PR DESCRIPTION
**Pull request by hlinnaka**
It has been unused since the 9.5 merge, which replaced the way GROUPING
SETS are planned and executed with the upstream implementation.

### CODE REVIEW

File: src/backend/cdb/cdbplan.c [hunk 1]
```diff
@@ -211,17 +211,6 @@ plan_tree_mutator(Node *node,
 			}
 			break;
 
-		case T_Repeat:
```
Comment (pengzhout) [inline src/backend/cdb/cdbplan.c]: T_Repeat seems to still be kept in node.h

File: src/include/executor/nodeRepeat.h [hunk 2]
```diff
@@ -1,25 +0,0 @@
...
- *-------------------------------------------------------------------------
- */
-
-#ifndef NODEREPEAT_H
-#define NODEREPEAT_H
-
-#include "nodes/execnodes.h"
-
-extern TupleTableSlot *ExecRepeat(RepeatState *repeatstate);
-extern RepeatState *ExecInitRepeat(Repeat *node, EState *estate, int eflags);
```
Comment (pengzhout) [inline src/include/executor/nodeRepeat.h]: structure RepeatState should also be removed.

Comment (pengzhout) [reply src/include/executor/nodeRepeat.h]: Add T_RepeatState.

**Comment by hlinnaka** [comment]
Thanks for the review @pengzhout, rebased and removed the stuff you pointed out. I'll push this after concourse has confirmed that everything is still green.

**Pull request merged**